### DISCUSSION TITLE
Cuda errors while trying to cmake nequip

### DISCUSSION THREAD

**Discussion by AbhayNag0** [mir-group/pair_nequip_allegro#31] [General]
Hello everyone,

I am getting some cuda errors while trying to cmake nequip. Nequip is not able to find cuda even though I am specifying the cuda path with `-DCUDA_TOOLKIT_ROOT_DIR`. What should I do?
These are the error messages I am getting:

-- Could NOT find CUDA (missing: CUDA_INCLUDE_DIRS CUDA_CUDART_LIBRARY) (found version "11.8")
CMake Warning at /home/arun/miniconda3/envs/nequip/lib/python3.8/site-packages/torch/share/cmake/Caffe2/public/cuda.cmake:31 (message):
  Caffe2: CUDA cannot be found.  Depending on whether you are building Caffe2
  or a Caffe2 dependent library, the next warning / error will give you more
  info.
Call Stack (most recent call first):
  /home/arun/miniconda3/envs/nequip/lib/python3.8/site-packages/torch/share/cmake/Caffe2/Caffe2Config.cmake:88 (include)
  /home/arun/miniconda3/envs/nequip/lib/python3.8/site-packages/torch/share/cmake/Torch/TorchConfig.cmake:68 (find_package)
  CMakeLists.txt:1017 (find_package)

CMake Error at /home/arun/miniconda3/envs/nequip/lib/python3.8/site-packages/torch/share/cmake/Caffe2/Caffe2Config.cmake:90 (message):
  Your installed Caffe2 version uses CUDA but I cannot find the CUDA
  libraries.  Please set the proper CUDA prefixes and / or install CUDA.
Call Stack (most recent call first):
  /home/arun/miniconda3/envs/nequip/lib/python3.8/site-packages/torch/share/cmake/Torch/TorchConfig.cmake:68 (find_package)
  CMakeLists.txt:1017 (find_package)

It would be great if someone can help me since I am new to this.

**Comment by Linux-cpp-lisp** [comment]
Hi @AbhayNag0 ,

Thanks for your interest in our code!

Do you have a complete CUDA installation including the developer tools (`nvcc` etc.)?

**Reply by AbhayNag0** [reply]
I actually needed to just use it with cpu. So I did a cpu only installation itself. Thanks a lot anyways.

**Reply by Linux-cpp-lisp** [reply]
Sounds good.  I can't find the issue right now but please note that performance is often significantly improved on CPU when PyTorch is built from source.
